module/Company/src/Company/Service/Company.php

```diff
@@ -246,6 +299,37 @@ public function insertCategoryByData($data, $files)
         return null;
     }
 
+    /**
+     * Checks if the data is valid, and if it is, inserts the label, and sets
+     * all data
+     *
+     * @param mixed $data
+     */
+    public function insertLabelByData($data, $files)
+    {
+        $labelForm = $this->getLabelForm();
+        $mergedData = array_merge_recursive(
+            $data->toArray(),
+            $files->toArray()
+        );
+        $labelForm->setData($mergedData);
+        $arr = [];
+        $labelForm->get('labels')->setObject($arr);
+        $valid = $labelForm->isValid();
+        if ($valid) {
+            $newLabels = $labelForm->getObject();
+            $id = -1;
+            foreach ($newLabels as $lang => $label) {
+                $arr[$lang] = $this->insertLabel($lang, $id, $label);
+                if ($id == -1) {
+                    $id = current($arr)->getId();
+                }
+            }
+            return $arr;
+        }
+        return null;
```

Comment:
Why not something like:

```
if (!$form->isValid()) {
    return null;
}
// do other things here
```

?

Reply:
This seems more logical, I also adjusted the existing code